### DISCUSSION TITLE
New UX for filtering

### DISCUSSION THREAD

**Discussion by kisst** [iann0036/former2#366] [Ideas]
When filtering for a given service at least two main linked services are causing a headache and a lot of manual cleanup on the generated code ( IAM  and Networking ) . 
However including them in the rendered IaC code isn't a good default either, as they could be input parameters ( CFN ) or just being located based on a given property ( TF data provider )   

My suggestion is  when filtered code render is requested, then the render should took three stages. 
1) Scan the account for the requested scope but follow the links references ( IAM roles, Network elements ) 
2) Render in a graph the scan result and let the user pick which linked resource to include,  exclude, or just refer to based on some property of the resource 
3) Generate the IaC based on the two perv steps

**Comment by kisst** [comment]
@iann0036 is this something what you would consider ? 
I was just about to create a quick export of a few IAM roles, I used the filter and the policies linked to the roles, where skipped, removed the filter and I have waaay too much stuff. There should be something between.

**Reply by iann0036** [reply]
Hey @kisst,

Just checking, is this in relation to the web UI or the CLI? Happy to consider it for the CLI, but would need to have a think about re-implementing the relationship graph resolution that the web UI's "Related Resources" does.

**Reply by kisst** [reply]
It's for CLI , I use former2 mainly from CloudShell .
